pyramid_oereb/contrib/print_proxy/mapfish_print/toc_pages.py

```diff
@@ -8,101 +8,158 @@
 class TocPages():
 
     def __init__(self, extract):
-        # variables taken from template toc.jrxml
-        self.disposable_height = 842 - 70  # A4 size - (footer + header)
-        self.d1_height = 77
-        self.d2_height = 29
-        self.d3_height = 61
-        self.d4_height = 44
-        self.d5_height = 15
-        self.d6_height = 90
-        self.d6_right_height = 23
-        self.d6_right_width = 233
-        self.d6_stuff_y_location = 39
-        self.d6_left_height = 0  # FIXME: compute this
-        self.title_size = 62
-        self.toc_title_height = 15 + 62 + 12  # height + location + item starting position
-        self.toc_item_height = 20
-        self.not_concerned_themes_title_height = 15 + 26  # height + location
-        self.not_concerned_themes_item_height = 12
-        self.theme_without_data_title_height = 12
-        self.theme_without_data_item_height = 12
+        self.total_height = 842
+        self.header_height = self.compute_header()
+        self.footer_height = self.compute_footer()
+        self.disposable_height = 842 - self.header_height - self.footer_height  # A4 size - (footer + header)
+        self.d1_height = 77  # toc.jrxml
+        self.d2_height = 29  # toc.jrxml
+        self.d3_height = 61  # toc.jrxml
+        self.d4_height = 44  # toc.jrxml
+        self.d5_height = 93  # toc.jrxml
+        self.d6_height = 38  # toc.jrxml
+        self.d6_left_height = 38  # toc.jrxml
+        self.d6_right_height = 20  # toc.jrxml
         self.extract = extract
+        self.display_qrcode = self.extract['Display_QRCode']
         self.total_length = self.compute_total_lenght()
 
+    def compute_header(self):
+        total_size = 0
+        page_top_margin = 28 # toc.jrxml
+        header_height = 60 # toc.jrxml
+        total_size += page_top_margin + header_height
+        log.debug(f"header total_size: {total_size}")
+        return total_size
+
+    def compute_footer(self):
+        total_size = 0
+        page_bottom_margin = 20 # toc.jrxml
+        footer_height = 10 # toc.jrxml
+        log.debug(f"header total_size: {total_size}")
```

Comment:
`total_size` is always zero.

Reply:
Well spotted, thanks! I just commited a bugfix.